source/opt/workaround_image_operands.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "workaround_image_operands.h"
+
+#include <list>
+#include <stack>
+
+namespace spvtools {
+namespace opt {
+
+Pass::Status DeanonymizeImages::Process(ir::IRContext* c) {
+  InitializeProcessing(c);
+  bool modified = false;
+  modified = FixupOpTypeImage();
+  return (modified ? Status::SuccessWithChange : Status::SuccessWithoutChange);
+}
+
+bool DeanonymizeImages::FixupOpTypeImage() {
+  auto* def_use_mgr = context()->get_def_use_mgr();
+  auto* type_mgr = context()->get_type_mgr();
+
+  bool modified = false;
+  auto* module = get_module();
+
+  std::vector<analysis::Image*> unknown_depth_images;
+
+  for (auto type : *type_mgr) {
+    // 2 means no indication as to whether this is a depth or non-depth image
+    if (auto* t = type.second->AsImage()) {
+      if (t->depth() == 2) {
+        unknown_depth_images.push_back(t);
+        modified = true;
+      }
+    }
+  }
+
+  std::vector<ir::Instruction*> fixed_instructions;
+
+  for (auto img_type : unknown_depth_images) {
+    // Find all of the places we used this image
+    def_use_mgr->ForEachUse(
```

Comment:
Similar to above, is using the def_use_mgr here the right solution, or should I just iterate all of the instructions instead.